plugins/command-manager/src/main/java/com/wazuh/commandmanager/rest/RestPostCommandAction.java

```diff
@@ -88,77 +86,59 @@ protected RestChannelConsumer prepareRequest(final RestRequest request, final No
+            throws IOException {
         log.info(
                 "Received {} {} request id [{}] from host [{}]",
                 request.method().name(),
                 request.uri(),
                 request.getRequestId(),
                 request.header("Host"));
-
-        /// Request validation
-        /// ==================
-        /// Fail fast.
+        // Request validation
         if (!request.hasContent()) {
-            // Bad request if body doesn't exist
             return channel -> {
                 channel.sendResponse(
                         new BytesRestResponse(RestStatus.BAD_REQUEST, "Body content is required"));
             };
         }
-
-        /// Request parsing
-        /// ===============
-        /// Retrieves and generates an array list of commands.
-        XContentParser parser = request.contentParser();
-        List<Command> commands = new ArrayList<>();
-        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-        // The array of commands is inside the "commands" JSON object.
-        // This line moves the parser pointer to this object.
-        parser.nextToken();
-        if (parser.nextToken() == XContentParser.Token.START_ARRAY) {
-            commands = Command.parseToArray(parser);
-        } else {
-            log.error("Token does not match {}", parser.currentToken());
+        List<Command> commands = getCommandList(request);
+        // Validate commands are not empty
+        if (commands.isEmpty()) {
+            return channel -> {
+                channel.sendResponse(
+                        new BytesRestResponse(
+                                RestStatus.BAD_REQUEST,
+                                "No valid commands detected in the request body."));
+            };
         }
-
-        /// Commands expansion
-        /// ==================
-        /// Transforms the array of commands to orders.
-        /// While commands can be targeted to groups of agents, orders are targeted to individual
-        // agents.
-        /// Given a group of agents A with N agents, a total of N orders are generated. One for each
-        // agent.
-        Documents documents = new Documents();
-        for (Command command : commands) {
-            Document document =
-                    new Document(
-                            new Agent(List.of("groups000")), // TODO read agent from wazuh-agents
-                            // index
-                            command);
-            documents.addDocument(document);
+        Orders orders = commandsToOrders(client, commands);
+        // Validate that the orders are not empty
+        if (orders.getOrders().isEmpty()) {
+            return channel -> {
+                channel.sendResponse(
+                        new BytesRestResponse(
+                                RestStatus.BAD_REQUEST,
+                                "Cannot generate orders. Invalid agent IDs or groups."));
```

Comment:
In which case an Agent ID is invalid?

Reply:
When the command with target type agent is received it could come with an invalid ID, or an ID that does not correspond to an indexed Agent